redbot/core/core_commands.py

```diff
@@ -336,7 +336,17 @@ async def info(self, ctx: commands.Context):
         except (aiohttp.ClientError, asyncio.TimeoutError):
             outdated = None
         else:
-            outdated = VersionInfo.from_str(data["info"]["version"]) > red_version_info
+            releases = data["releases"]
+            valid_releases = sorted(
+                [
+                    r
+                    for i in releases.keys()
+                    if (r := VersionInfo.from_str(i)) and r.releaselevel == VersionInfo.FINAL
+                ],
+                reverse=True,
+            )
+            pypi_version = valid_releases[0] if valid_releases else None
+            outdated = pypi_version and pypi_version > red_version_info
```

Comment:
Could we perhaps put this in some sort of internal util function, possibly including the call to PyPI's site as well, i.e. `fetch_latest_red_version_info() -> Optional[VersionInfo]`? This is getting lengthy and seems like an unnecessary code repetition.